src/main/java/org/testobject/appium/testng/TestObjectTestNGTestResultWatcher.java

```diff
@@ -16,11 +15,17 @@
 	public void onTestStart(ITestResult testResult) {
 		super.onTestStart(testResult);
 
-		if (testResult.getInstance() instanceof TestObjectWatcherProvider) {
-			TestObjectWatcherProvider provider = (TestObjectWatcherProvider) testResult.getInstance();
-			URL apiEndpoint = provider.getApiEndpoint();
-			reporter = new IntermediateReporter(apiEndpoint, provider.getIsLocalTest());
-			reporter.setRemoteWebDriver(provider.getDriver());
+		Object instance = testResult.getInstance();
+
+		if (instance instanceof TestObjectWatcherProvider) {
+
+			TestObjectWatcherProvider watcherProvider = ((TestObjectWatcherProvider) instance);
+			TestObjectTestNGTestProvider provider = watcherProvider.getProvider();
+
+			URL apiEndpoint = provider.getEndpoint();
+			reporter = new IntermediateReporter(apiEndpoint, provider.isLocalTest());
+			reporter.setRemoteWebDriver(provider.getRemoteWebDriver());
```

Comment:
this doesn't work out. you need to handover the TestObjectTestNGTestProvider and whenever you want to really use the RemoteWebDriver you have to call getRemoveWebDriver()

It could actually be that in that phase of the test the webdriver is not being initialised.